File: solutions/1493. Longest Subarray of 1's After Deleting One Element/1493.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  int longestSubarray(const vector<int>& nums) {
    int l = 0;
    int zeros = 0;

    for (const int num : nums) {
      if (num == 0)
        ++zeros;
      if (zeros > 1 && nums[l++] == 0)
        --zeros;
    }

    return nums.size() - l - 1;
  }
};

File: solutions/1493. Longest Subarray of 1's After Deleting One Element/1493.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def longestSubarray(self, nums: List[int]) -> int:
    l = 0
    zeros = 0

    for num in nums:
      if num == 0:
        zeros += 1
      if zeros > 1:
        if nums[l] == 0:
          zeros -= 1
        l += 1

    return len(nums) - l - 1